pkg/services/ngalert/api/authorization.go

```diff
@@ -125,20 +125,34 @@ func (api *API) authorize(method, path string) web.Handler {
 	// Silences. External AM.
 	case http.MethodDelete + "/api/alertmanager/{DatasourceID}/api/v2/silence/{SilenceId}":
 		eval = ac.EvalPermission(ac.ActionAlertingInstancesExternalWrite, datasources.ScopeProvider.GetResourceScope(ac.Parameter(":DatasourceID")))
+	case http.MethodDelete + "/api/alertmanager/uid/{DatasourceUID}/api/v2/silence/{SilenceId}":
+		eval = ac.EvalPermission(ac.ActionAlertingInstancesExternalWrite, datasources.ScopeProvider.GetResourceScope(ac.Parameter(":DatasourceUID")))
 	case http.MethodPost + "/api/alertmanager/{DatasourceID}/api/v2/silences":
 		eval = ac.EvalPermission(ac.ActionAlertingInstancesExternalWrite, datasources.ScopeProvider.GetResourceScope(ac.Parameter(":DatasourceID")))
+	case http.MethodPost + "/api/alertmanager/uid/{DatasourceUID}/api/v2/silences":
```

Comment:
If we do this, are we stuck with this `/uid` component in the path forever? It's a bit unusual in terms of URL paths. If we want a path component, what about `/datasource`? That names the object that the ID is identifying. 

Our UID generation algorithm also guarantees that a UID will never be a simple integer string, so we could consider just parsing the `{DatasourceID}` string and seeing if it's a valid integer.

I slightly prefer the path parameter `alertmanager/datasource/{UID}`

(I have the same comment on your other PRs, so let's have the conversation here)

Reply:
I really like `alertmanager/datasource/{UID}` moving forward.

Reply:
> Our UID generation algorithm also guarantees that a UID will never be a simple integer string, so we could consider just parsing the {DatasourceID} string and seeing if it's a valid integer.

datasource UIDs are not always generated but they can be also user defined, for example [in case of provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/?plcmt=footer#example-data-source-config-file).
Therefore, there are some cases where it's hard to decide if the user intends to query using the sequential ID or the UID as discussed [here](https://github.com/grafana/grafana/pull/47634#discussion_r848389970).

Reply:
So what do you think about replacing `uid` in this path with `datasource`?

Reply:
I definitely can do that but there is also an idea [of removing completely the use of sequential datasource IDs from the Grafana Alerting API](https://raintank-corp.slack.com/archives/C028MCV4R7C/p1650978838470629?thread_ts=1650557233.740419&cid=C028MCV4R7Chttps://raintank-corp.slack.com/archives/C028MCV4R7C/p1650978838470629?thread_ts=1650557233.740419&cid=C028MCV4R7C) and support only UIDs (but it requires modifications in the UI as well) so let's decide first on that.